File: 0001-two-sum/0001-two-sum.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        num_map = {}  # Dictionary to store number and its index
        
        for i, num in enumerate(nums):
            complement = target - num  # Calculate the complement
            if complement in num_map:
                return [num_map[complement], i]  # Return indices of the two numbers
            num_map[num] = i  # Store the index of the current number
        
        return []  # Return empty list if no solution is found

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<pair<int,int>> temp;
        int n = nums.size();
        for(int i = 0;i<n;i++){
            temp.push_back({nums[i],i});
        }
        sort(temp.begin(),temp.end());
        int left = 0;
        int right = n-1;
        while(left<right){
            int sum = temp[left].first+temp[right].first;
            if(sum == target){
                return {temp[left].second,temp[right].second};
            }
            else if(sum>target){
                right--;
            }
            else if(sum<target){
                left++;
            }
        }
            return {};
    }

};